packages/pwa/app/utils/utils.js

```diff
@@ -162,3 +164,107 @@ export const capitalize = (text) => {
  * @returns {object} - url object from the pwa-kit-config.json file
  */
 export const getUrlConfig = () => pwaKitConfig?.app?.url
+/**
+ * A util to return current multi site configuration
+ * @returns {object} - a list of site objects from the pwa-kit-config.json file
+ */
+export const getSitesConfig = () => pwaKitConfig?.app?.sites
+/**
+ * A util to return current defaultSiteId
+ * @returns {string} - default site Id string from pwa-kit-config file
+ */
+export const getDefaultSiteId = () => pwaKitConfig?.app?.defaultSiteId
+
+/**
+ * A util to get the siteId
+ * @param url
+ * @returns {string}
+ */
+export const getSiteId = (url) => {
+    let path = url
+    if (!path) {
+        path = `${window?.location.pathname}${window?.location.search}`
+    }
+
+    const {hostname, pathname, search} = new URL(`${getAppOrigin()}${path}`)
+    let siteId = getSiteIdByHostname(hostname)
+
+    if (siteId) {
+        return siteId
+    }
+
+    siteId = getSiteIdByAlias(`${pathname}${search}`)
+    return siteId
+}
+
+/**
+ * Get the site Id based on the given hostname
+ * @param {string} hostname
+ * @returns {string} siteId
+ */
+export const getSiteIdByHostname = (hostname) => {
+    const sitesConfig = getSitesConfig()
+    if (!sitesConfig.length) throw new Error('No site config found. Please check you configuration')
+    if (!hostname) return undefined
+
+    const site = sitesConfig.filter((site) => {
+        return site?.hostname?.some((i) => i.includes(hostname))
+    })
+
+    return site?.length === 1 ? site[0].id : undefined
+}
+
+/**
+ * get the site id based on the site alias in the given url
+ * @param {string} url - input url
+ * @returns {string} siteId
+ */
+export const getSiteIdByAlias = (url) => {
```

Comment:
The name of this function is a little wonky.. the pattern for these types of getter functions where the name is something like `getThisByThat` is that you are usually using `That` is a parameter. Here is an example:

```
const el = document.getElementById('id')
```

So following that pattern the argument in your function should be `alias`. Make sense?

Reply:
Another issue that I foresee, is that we technically don't know if the value in the URL is a `siteId` or a `siteAlias` that is the exact reason why we use the name `site` for it. For that reason we have to create a type of resolution contract where we can document how we come up with the site matching algorithm. 

For example, our site matcher might do something like, first look for the a site starting at the beginning of the array and if there the site matches the alias we return that site as the match, if there is no site with an alias that matches the `site` value you are trying to match, then it will match by the `siteId` or `id` property. 

There are a few assumptions that we have to make, 1 is that there are no duplicate alias's, and also that id's are distinct. 

Another thing that will throw a spanner in the works, is that from looking at your deployed examples it doesn't look like we are showing the site or locale when it's the default, that will all works into how we are parting the url I suppose. Something to think about.